iothub/device/src/Pipeline/RetryDelegatingHandler.cs

```diff
@@ -137,18 +141,17 @@ await _internalRetryHandler
                         async () =>
                         {
                             await VerifyIsOpenAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
I'll renew my dislike for the semaphore wait to evaluate _isOpen. I don't see what it buys us, and it costs each operation an extra semaphore get.

Did you have a future plan to refactor that?

Reply:
Yess, we'll get rid of the semaphores once we introduce the atomic variable for the state variable